api/v1beta1/awsloadbalancercontroller_types.go

```diff
@@ -0,0 +1,206 @@
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Config *AWSLoadBalancerDeploymentConfig `json:"config,omitempty"`
+
+	// AWSAddon describes the AWS services that can be integrated with
+	// the AWS Load Balancer.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	EnabledAddons []AWSAddon `json:"enabledAddons,omitempty"` // indicates which AWS addons should be disabled.
+
+	// Credentials is a reference to a secret containing
+	// the AWS credentials to be used by the controller.
+	// The secret is required to be in the operator namespace.
+	// If this field is empty - the credentials will be
+	// requested using the Cloud Credentials API,
+	// see https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/about-cloud-credential-operator.html.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Credentials *SecretReference `json:"credentials,omitempty"`
+}
+
+type AWSLoadBalancerDeploymentConfig struct {
+
+	// +kubebuilder:default:=2
+	// +kubebuilder:validation:Optional
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+}
+
+// SecretReference contains the information to let you locate the desired secret.
+// Secret is required to be in the operator namespace.
+type SecretReference struct {
+	// Name is the name of the secret.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// AWSLoadBalancerControllerStatus defines the observed state of AWSLoadBalancerController.
+type AWSLoadBalancerControllerStatus struct {
+
+	// Conditions is a list of operator-specific conditions
+	// and their status.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the most recent generation observed.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Subnets contains details of the subnets of the cluster
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Subnets *AWSLoadBalancerControllerStatusSubnets `json:"subnets,omitempty"`
+
+	// IngressClass is the current default Ingress class.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	IngressClass string `json:"ingressClass,omitempty"`
+}
+
+type AWSLoadBalancerControllerStatusSubnets struct {
+	// SubnetTagging indicates the current status of the subnet tags
+	// +kubebuilder:validation:Optional
+	// +optional
+	SubnetTagging SubnetTaggingPolicy `json:"subnetTagging,omitempty"`
+
+	// Internal is the list of subnet ids which have the tag `kubernetes.io/role/internal-elb`
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Internal []string `json:"internal,omitempty"`
+
+	// Public is the list of subnet ids which have the tag `kubernetes.io/role/elb`
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Public []string `json:"public,omitempty"`
```

Comment:
Will every public subnet owned/reconciled by the lb-controller be always tagged with `kubernetes.io/role/elb`? Is that a hardcoded tag? Or, the tag is independent of the lb-controller?

Reply:
> Will every public subnet owned/reconciled by the lb-controller be always tagged with kubernetes.io/role/elb?

Yes. This is a tag to say that internet facing load balancer can be created here.

> Is that a hardcoded tag?

Yes but this one is not put by the openshift-installer like the private one. For some reason the tagging of the public subnets with ELB role was removed from the installer at some point. We tried to clarify it but didn't manage to change this with the installer team. That's why the operator now does this job.

Reply:
As discussed, I added more comments to the `status.subnets` to clarify which tags are used to classify the subnets.